crates/matching-engine/src/lib.rs

```diff
@@ -35,3 +45,36 @@ pub fn build_book(
 
     OrderBook::new(id, amm, bids, asks, Some(book::sort::SortStrategy::ByPriceByVolume))
 }
+
+pub async fn configure_uniswap_manager<T: Transport + Clone, N: Network>(
+    provider: Arc<impl Provider<T, N>>,
+    state_notification: CanonStateNotifications,
+    uniswap_pool_registry: UniswapPoolRegistry,
+    current_block: BlockNumber
+) -> UniswapPoolManager<CanonicalStateAdapter, DataLoader<PoolId>, PoolId> {
+    let mut uniswap_pools: Vec<_> = uniswap_pool_registry
+        .pools()
+        .keys()
+        .map(|pool_id| {
+            let initial_ticks_per_side = 200;
+            EnhancedUniswapPool::new(
+                DataLoader::new_with_registry(*pool_id, uniswap_pool_registry.clone()),
+                initial_ticks_per_side
+            )
+        })
+        .collect();
+
+    for pool in uniswap_pools.iter_mut() {
+        pool.initialize(Some(current_block), provider.clone())
+            .await
+            .unwrap();
+    }
```

Comment:
Covert above into a stream and then just do this operation at the same time to avoid the double loop.